### PR TITLE
[embind] Minor refactor to use more modern JS features

### PR DESCRIPTION
**Pull request by sbc100**

### CODE REVIEW

File: src/embind/embind_shared.js [hunk 1]
```diff
@@ -265,8 +268,8 @@ var LibraryEmbindShared = {
       for (var i = isClassMethodFunc?1:2; i < argTypes.length; ++i) { // Skip return value at index 0 - it's not deleted here. Also skip class type if not a method.
         var paramName = (i === 1 ? "thisWired" : ("arg"+(i - 2)+"Wired"));
         if (argTypes[i].destructorFunction !== null) {
-          invokerFnBody += paramName+"_dtor("+paramName+");\n";
-          args1.push(paramName+"_dtor");
+          invokerFnBody += `${paramName}_dtor${paramName});\n`;
```
Comment (brendandahl) [inline src/embind/embind_shared.js:271]: Looks like a paren was dropped here.

**Review by brendandahl — CHANGES_REQUESTED** [review]

File: src/embind/embind_shared.js (hunk 1 above)
Comment (sbc100) [reply src/embind/embind_shared.js]: Oops, good catch

File: src/embind/embind_shared.js [hunk 2]
```diff
@@ -288,7 +291,7 @@ var LibraryEmbindShared = {
     invokerFnBody += "return Asyncify.currData ? Asyncify.whenDone().then(onDone) : onDone(" + (returns ? "rv" : "") +");\n"
 #elif ASYNCIFY == 2
     invokerFnBody += "}\n";
-    invokerFnBody += "return " + (isAsync ? "rv.then(onDone)" : "onDone(" + (returns ? "rv" : "") + ")") + ";";
+    invokerFnBody += "return " + (isAsync ? "rv.then(onDone)" : `onDone(${returnVal})` + ";";
```
Comment (brendandahl) [inline src/embind/embind_shared.js:294]: Looks like another `)` was dropped here.

**Review by brendandahl — APPROVED** [review]
r+ with the tests passing (hopefully last comment fixes them)

**Pull request merged**